src/main/java/org/ballerinalang/command/util/ToolUtil.java

```diff
@@ -110,6 +129,49 @@ public static String getCurrentBallerinaVersion() {
         }
     }
 
+    /**
+     * Sets custom SSL context for update tool.
+     */
+    public static void setCustomSSLContext() throws IOException {
+        // Load custom truststore if provided, otherwise use the default truststore
+        try {
+            KeyStore truststore;
+            if (trustStorePath != null && trustStorePassword != null) {
+                truststore = KeyStore.getInstance(KeyStore.getDefaultType());
+                try (InputStream keys = Files.newInputStream(Paths.get(trustStorePath))) {
+                    truststore.load(keys, trustStorePassword.toCharArray());
+                }
+            } else {
+                truststore = KeyStore.getInstance(KeyStore.getDefaultType());
+                try (InputStream defaultKeys = Files.newInputStream(Paths.get(System.getProperty("java.home") +
```

Comment:
Shall we use constants for "java.home", "/lib/security/cacerts", "changeit" etc?

Reply:
@Thevakumar-Luheerathan Addressed suggestions